onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1044,6 +1044,10 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BeamSearch, 1,
                                 .Attr("early_stopping", "early stop or not", AttributeProto::INT, static_cast<int64_t>(0))
                                 .Attr("model_type", "model type: 0 for GPT-2; 1 for encoder decoder like T5", AttributeProto::INT, static_cast<int64_t>(0))
                                 .Attr("encoder", "The subgraph for initialization of encoder and decoder. It will be called once before decoder subgraph.", AttributeProto::GRAPH, OPTIONAL_VALUE)
+                                .Attr("init_decoder",
```

Comment:
maybe we can leverage the encoder_decoderinit so there's no need to add another attribute?

Reply:
You mean in the conversion tool or here in the op schema where-in we can just re-use the existing attribute or both ? 
Open to any such change. Not sure if using "encoder" in GPT2 was semantically right. Please advise.